pkg/storage/secret/secure_value_store.go

```diff
@@ -28,3 +46,209 @@ func ProvideSecureValueStorage(db db.DB, cfg *setting.Cfg, features featuremgmt.
 type storage struct {
 	db db.DB
 }
+
+func (s *storage) Create(ctx context.Context, sv *secretv0alpha1.SecureValue) (*secretv0alpha1.SecureValue, error) {
+	authInfo, ok := claims.From(ctx)
+	if !ok {
+		return nil, fmt.Errorf("missing auth info in context")
+	}
+
+	// This should come from the keeper. From this point on, we should not have a need to read value/ref.
+	externalID := "TODO"
+	sv.Spec.Value = ""
+	sv.Spec.Ref = ""
+
+	row, err := toCreateRow(sv, authInfo.GetUID(), externalID)
+	if err != nil {
+		return nil, fmt.Errorf("to create row: %w", err)
+	}
+
+	err = s.db.WithTransactionalDbSession(ctx, func(sess *sqlstore.DBSession) error {
```

Comment:
Does this need to be a transaction at the moment?

Reply:
@PoorlyDefinedBehaviour you're right, removed!